.github/workflows/release.yml

```diff
@@ -850,6 +850,189 @@ jobs:
+      with:
+        name: openttd-windows-x86
+
+    - name: Download x64 build
+      uses: actions/download-artifact@v2
+      with:
+        name: openttd-windows-x64
+
+    - name: Download arm64 build
+      uses: actions/download-artifact@v2
+      with:
+        name: openttd-windows-arm64
+
+    - name: Unpack builds
+      shell: bash
+      run: |
+        mkdir builds
+        cd builds
+
+        function extract {
+            mkdir $1
+
+            # Extract the zip version of the release
+            unzip ../openttd-*-windows-$2.zip -d $1
+
+            # Remove the extraneous directory
+            mv $1/openttd-*-windows-$2/* $1/
+            rmdir $1/openttd-*-windows-$2
+
+            # Move the openttd.exe to the '{arch}-binaries' folder
+            mkdir $1-binaries
+            mv $1/openttd.exe $1-binaries/
+        }
+
+        extract x86 win32
+        extract x64 win64
+        extract arm64 arm64
+
+        # Use the "x86" folder as the source of the common binaries (lang files, etc) and remove the others
+        mv x86 common-binaries
+        rm -rf x64 arm64
+
+    - name: Install OpenGFX
+      shell: bash
+      run: |
+        mkdir -p builds/common-binaries/baseset
+        cd builds/common-binaries/baseset
+
+        echo "::group::Download OpenGFX"
+        curl -L https://cdn.openttd.org/opengfx-releases/7.1/opengfx-7.1-all.zip -o opengfx-all.zip
+        echo "::endgroup::"
+
+        echo "::group::Unpack OpenGFX"
+        unzip opengfx-all.zip
+        tar xf opengfx-*.tar
+        echo "::endgroup::"
+
+        rm -f opengfx-all.zip opengfx-*.tar
+
+    - name: Install OpenMSX
+      shell: bash
+      run: |
+        mkdir -p builds/common-binaries/baseset
+        cd builds/common-binaries/baseset
+
+        echo "::group::Download OpenMSX"
+        curl -L https://cdn.openttd.org/openmsx-releases/0.4.2/openmsx-0.4.2-all.zip -o openmsx-all.zip
+        echo "::endgroup::"
+
+        echo "::group::Unpack OpenGFX"
+        unzip openmsx-all.zip
+        tar xf openmsx-*.tar
+        echo "::endgroup::"
+
+        rm -f openmsx-all.zip openmsx-*.tar
+
+    - name: Install OpenSFX
+      shell: bash
+      run: |
+        mkdir -p builds/common-binaries/baseset/opensfx
+        cd builds/common-binaries/baseset/opensfx
+
+        echo "::group::Download OpenSFX"
+        curl -L https://cdn.openttd.org/opensfx-releases/1.0.3/opensfx-1.0.3-all.zip -o opensfx-all.zip
+        echo "::endgroup::"
+
+        echo "::group::Unpack OpenSFX"
+        unzip opensfx-all.zip
+        tar xf opensfx-*.tar
+        echo "::endgroup::"
+
+        rm -f opensfx-all.zip opensfx-*.tar
+
+    - name: Generate signing certificate
+      shell: cmd
+      run: |
+        cd builds
+
+        REM We need to provide a signed .appx to the Windows Store, so generate a certificate with password 'password'
+        call ..\os\windows\winstore\generate-key.bat "CN=78024DA8-4BE4-4C77-B12E-547BBF7359D2" password cert.pfx
```

Comment:
So we generate a new certificate every time, to sign a file to send to Microsoft .. that sounds ... useful? Lol .. I am sure Microsoft has its reasons ...

Reply:
Ugh, hit enter too quickly: should the CN number be private? (read: in a GitHub Secret?)

Reply:
We generate a new certificate each time, though we could alternatively manually generate a certificate and use that every time (which might be easier if anybody wanted to test the .appx outside the Store, in that they'd just need to trust that certificate once). Our standard code signing certificate however doesn't work here because the app has to have the CN that the Store gave us. Microsoft then discards our certificate and signs it with their own certificate upon upload to the store.

In terms of the CN being secret, no, it doesn't need to be; anybody can see it by installing the app from the store and poking around in the manifest files that get installed to your machine.